### DISCUSSION TITLE
Understanding high memory usage of otel grpc collector

### DISCUSSION THREAD

**Discussion by shravani154** [open-telemetry/opentelemetry-collector#8681] [Q&A]
Hello team. I'm currently utilizing Otel Java Agent version 1.30.0 with a gRPC collector for trace collection. At the moment, I'm not exporting metrics and logs. However, I've observed a gradual increase in memory usage by the Otel collector, starting at 400MB and gradually reaching up to 1300MB or more, without any reduction. I'm interested in finding solutions to address this issue.

**Comment by amankhare94** [comment]
@shravani154  facing similar issue, did you find any solution for this?

**Reply by thiruvikraman-subburaman** [reply]
Ensure you are not creating grpc connection frequently for every event as grpc connections are meant to keep longlive connection by design.

**Reply by cforce** [reply]
So we need best practice configuration  for grpc connection and package handling

**Comment by lazy12737** [comment]
+1 why
